### PR TITLE
avoid explicit calls to Dispose() in favor of 'using' statement

### PR DESCRIPTION
**Pull request by jgilbert2017**
avoid explicit calls to Dispose() in favor of 'using' statement.  this ensures Dispose() is called in the event of an exception and improves code readability.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10694)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridViewLinkCell.cs [hunk 1]
```diff
@@ -1038,14 +1038,14 @@ protected override void OnMouseUp(DataGridViewCellMouseEventArgs e)
             bool isActive = (LinkState & LinkState.Active) == LinkState.Active;
 
             LinkUtilities.EnsureLinkFonts(cellStyle.Font, LinkBehavior, ref getLinkFont, ref getHoverFont, isActive);
+            using Font linkFont = getLinkFont;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridViewLinkCell.cs:1041]: Why did you move these lines up, before the call to `ComputeTextFormatFlagsForCellStyleAlignment`?

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/TreeView/TreeView.cs [hunk 2]
```diff
@@ -2736,11 +2736,8 @@ private unsafe void CustomDraw(ref Message m)
                 }
                 else if (_drawMode == TreeViewDrawMode.OwnerDrawAll)
                 {
-                    Graphics g = nmtvcd->nmcd.hdc.CreateGraphics();
-
                     DrawTreeNodeEventArgs e;
-
-                    try
+                    using (Graphics g = nmtvcd->nmcd.hdc.CreateGraphics())
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/TreeView/TreeView.cs:2740]: Is it possible to use a using statement here as well?

File: src/System.Windows.Forms/src/System/Windows/Forms/Design/ComponentEditorForm.cs [hunk 3]
```diff
@@ -202,15 +202,15 @@ private void OnConfigureUI()
 
         if (_pageSites is not null)
         {
+            using Graphics graphics = CreateGraphicsInternal();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Design/ComponentEditorForm.cs:205]: We were not creating graphics when _pageSites.Length was zero previously.

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridViewLinkCell.cs (hunk 1 above)
Comment (jgilbert2017) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridViewLinkCell.cs:1041]: to ensure the using statement takes effect prior to the call to `ComputeTextFormatFlagsForCellStyleAlignment`.  if `ComputeTextFormatFlagsForCellStyleAlignment` throws we need to dispose.

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/TreeView/TreeView.cs (hunk 2 above)
Comment (jgilbert2017) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/TreeView/TreeView.cs:2740]: this ensures the earliest possible disposal of the `g`.  using statement doesn't dispose until the end of the block.

File: src/System.Windows.Forms/src/System/Windows/Forms/Design/ComponentEditorForm.cs (hunk 3 above)
Comment (jgilbert2017) [reply src/System.Windows.Forms/src/System/Windows/Forms/Design/ComponentEditorForm.cs:205]: nice catch, fixed.

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridViewLinkCell.cs (hunk 1 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridViewLinkCell.cs:1041]: I see, it's allocated in EnsureLinkFonts

**Review by Tanya-Solyanik — APPROVED** [review]

**Review by Tanya-Solyanik — APPROVED** [review]

**Review by lonitra — APPROVED** [review]
